src/fields/field_formula.cc

```diff
@@ -44,9 +45,17 @@ const Input::Type::Record & FieldFormula<spacedim, Value>::get_input_type()
                                         " - array of strings of size (($N$)) to enter diagonal matrix\n"
                                         " - array of strings of size (($\\frac12N(N+1)$)) to enter symmetric matrix (upper triangle, row by row)\n"
                                         " - just one string to enter (spatially variable) multiple of the unit matrix.\n"
-                                        "Formula can contain variables ```x,y,z,t``` and usual operators and functions." )
+                                        "Formula can contain variables ```x,y,z,t,d``` and usual operators and functions." )
 			//.declare_key("unit", FieldAlgorithmBase<spacedim, Value>::get_input_type_unit_si(), it::Default::optional(),
 			//							"Definition of unit.")
+			.declare_key("surface_direction", it::String(), it::Default("\"0 0 1\""),
+										"The vector used to project evaluation point onto the surface.")
+			.declare_key("surface_region", it::String(), it::Default::optional(),
+										"The name of region set considered as the surface. You have to set surface region if you "
+										"want to use formula variable ```d```.")
+			.declare_key("max_depth", it::Double(0.0), it::Default("1e06"),
```

Comment:
Why such limit? Not very useful. To deal with failed projections we should 
implement sort of projection in BIH  to find nearest box. Then we just take e.g. barycenter of corresponding triangle. If to far we can report an error. Alternatively we can introduce a key for default value of 'd'.